src/torch_models.py

```diff
@@ -352,12 +353,31 @@ class ImplicitMLPRegressor(PyTorchRegressor):
     coming soon.
     """
 
-    def __init__(self, seed: int, hid_sizes: List[int], max_train_iters: int,
-                 clip_gradients: bool, clip_value: float, learning_rate: float,
+    def __init__(self,
+                 seed: int,
+                 hid_sizes: List[int],
+                 max_train_iters: int,
+                 clip_gradients: bool,
+                 clip_value: float,
+                 learning_rate: float,
                  num_samples_per_inference: int,
-                 num_negative_data_per_input: int, temperature: float) -> None:
+                 num_negative_data_per_input: int,
+                 temperature: float,
+                 inference_method: str,
+                 derivate_free_num_iters: Optional[int] = None,
+                 derivate_free_sigma_init: Optional[float] = None,
+                 derivate_free_shrink_scale: Optional[float] = None) -> None:
         super().__init__(seed, max_train_iters, clip_gradients, clip_value,
                          learning_rate)
+        assert inference_method in ("sample_once", "derivate_free")
```

Comment:
can remove this assert too, you'll hit the exception below

Reply:
i'm confused, did you comment on the same line twice somehow?